### ISSUE TITLE
metrics incorrect when filtering active

### ISSUE THREAD

**Issue by plastikfan** [snivilised/extendio#94]
may be caused by this code:

```golang
			decorator := &LabelledTraverseCallback{
				Label: "filter decorator",
				Fn: func(item *TraverseItem) *LocalisableError {
					if frame.filters.Node.IsMatch(item) {
						return decorated.Fn(item)
					}
					return nil
				},
			}
```

which would be fixed by:

```golang
			decorator := &LabelledTraverseCallback{
				Label: "filter decorator",
				Fn: func(item *TraverseItem) *LocalisableError {
					if frame.filters.Node.IsMatch(item) {
						return decorated.Fn(item)
					} else {
						item.skip = true
					}
					return nil
				},
			}
```

ie, when item does not match filter, the the skip flag must be set

**Comment by plastikfan** [comment]
OOPS I f**ked up the commit for this. Instead of creating a new commit as per normal, I amended the previous commit which was a merge for a different issue(merge branch ref/rename-api-items)  so the commit tree now looks weird at this point.

**Comment by plastikfan** [comment]
The commi for this fix is a6a2271
